core/src/main/scala/geotrellis/store/GeoTrellisRasterSourceLegacy.scala

```diff
@@ -17,44 +17,102 @@
 package geotrellis.store
 
 import geotrellis.proj4._
-import geotrellis.raster.io.geotiff.{Auto, AutoHigherResolution, Base, OverviewStrategy}
+import geotrellis.raster.io.geotiff.{AutoHigherResolution, OverviewStrategy}
 import geotrellis.raster.resample.{NearestNeighbor, ResampleMethod}
 import geotrellis.raster._
 import geotrellis.layer._
+import geotrellis.layer.filter._
 import geotrellis.vector._
 
+import java.time.ZonedDateTime
+
+case class LayerLegacy(id: LayerId, metadata: TileLayerMetadata[_], bandCount: Int) {
```

Comment:
Since we have GeoTrellisLayers here, why not to keep all the changes here and after approval port them into GeoTrellis core